sensor/common/sensor/central_communication_impl.go

```diff
@@ -242,7 +247,41 @@ func (s *centralCommunicationImpl) initialSync(stream central.SensorService_Comm
 		return err
 	}
 
-	return s.initialPolicySync(stream, detector)
+	if err := s.initialPolicySync(stream, detector); err != nil {
+		return err
+	}
+
+	if s.clientReconciliation {
+		// If client is reconciling, Sensor needs to wait for Central to send the hashes
+		return s.initialDeduperSync(stream, s.syncTimeout)
+	}
+	return nil
+}
+
+func (s *centralCommunicationImpl) initialDeduperSync(stream central.SensorService_CommunicateClient, timeout time.Duration) error {
```

Comment:
I would suggest to remove this block from this PR in favor or the change here #7755. If you want to test with the actual fetching of deduper state, let's rebase this branch on top of the 7755 one.

Reply:
Done in #8084